lib/iris/experimental/ugrid/__init__.py

```diff
@@ -3469,7 +3606,17 @@ def identify(cls, variables, ignore=None, target=None, warn=True):
         log_level = logging.WARNING if warn else logging.DEBUG
 
         # Identify all CF-UGRID mesh variables.
+        all_vars = target == variables
         for nc_var_name, nc_var in target.items():
+            if all_vars:
+                # SPECIAL BEHAVIOUR FOR MESH VARIABLES.
+                # We are looking for all mesh variables. Check if THIS variable
+                #  is a mesh using its own attributes.
+                if getattr(nc_var, "cf_role", "") == "mesh_topology":
+                    result[nc_var_name] = CFUGridMeshVariable(
```

Comment:
This is useful, but there are now 2 routes which lead to a CFUGridMeshVariable being constructed : Here, and below in line 3641.
Which is fine, given that..
 (a) id via a link-to is (now) no longer required to id a mesh in a given file (e.g.  for in a meshfile, or if it got missed), which we absolutely now want, _and_
 (b) we will also (now) tolerate a missing or miss-spelt 'cf_role' and still build a mesh
 
But.. I think this code will now, in the 'normal' case, be creating every mesh twice and setting `result[mesh-var-name]` twice.  
So it would be good to **check for a pre-existing entry in result[same-name]** and (a) check it matches (b) not re-create and over-write

**Also**, when creating "from a link", we should now be **checking `cf_role` and warning if it is not good**.
(i.e. around line 3641)

Reply:
#### Creating meshes twice

This is already the case for any variables that are referenced more than once e.g. files that have many data variables but which share coordinates/meshes/etc. If this hasn't been considered a problem so far in Iris' life, I'm reluctant to confuse the codebase by making a specific allowance here. Of course if it _has_ always been a problem we ought to fix it for every `identify()` method.

#### Bad `cf_role` warning/tolerance

Excellent point. I've mirrored what you did with `topology_dimension` in terms of warning and correcting the loaded value: 48c6d145b9ccfb144c9d67b1765c5e792e942fd6